app/assets/javascripts/application.js

```diff
@@ -23,4 +23,13 @@ $(document).on("turbolinks:load", function() {
       minDate: '+0d'
     });
   });
-});
\ No newline at end of file
+
+  // Used when selecting a set (create work order step 1) to allow a radio button
+  // to be checked by clicking the row, instead of just the radio button
+  $('#set-list-table tr').click(function (event) {
+    if (event.target.type !== 'radio') {
+      $(':radio', this).trigger('click');
+    }
+  })
```

Comment:
This solution avoids the problem of re-triggering the clicking event by filtering the elements that are checkboxes, to not trigger the click on them and end up in an infinite loop. I'm thinking, that maybe you don't need to trigger the event at all, just update the checking status of the checkbox (setting the `checked` attribute to true, or false), so you won't have to filter the checkboxes because it won't re-trigger a `click()` event. It would also keep the solution simpler.